appstore/validator.go

```diff
@@ -120,7 +121,20 @@ func (c *Client) Verify(req IAPRequest, result interface{}) error {
 	}
 	defer resp.Body.Close()
 
-	err = json.NewDecoder(resp.Body).Decode(result)
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	r := &IAPResponse{}
+	err = json.Unmarshal(bodyBytes, r)
+	if err == nil && r.Status == 21007 {
```

Comment:
Definitely want a comment here explaining why we're doing this, ideally with a link to Apple's docs.